include/h2o/httpclient.h

```diff
@@ -328,6 +328,10 @@ typedef struct st_h2o_httpclient__h2_conn_t {
      * `num_streams`.
      */
     h2o_linklist_t link;
+    /**
+     * close the connection
+     */
+    int (*close_conn)(struct st_h2o_httpclient__h2_conn_t *conn);
```

Comment:
I'm not sure if we need to use a callback when the object that we are dealing is already an HTTP/2-specific type.

Maybe create an external function like `h2o_httpclient__h2_initiate_close`?